src/common/snippets/tests/src/pass/precision_propagation.cpp

```diff
@@ -0,0 +1,319 @@
+    std::pair<Shape, Shape> shapes;
+    PrecisionPropagationParamsValues test_values;
+    std::tie(shapes, test_values) = obj.param;
+
+    auto to_string = [](const std::set<std::vector<element::Type>>& precisions_pack) noexcept {
+        auto to_string = [](const std::vector<element::Type>& precisions) noexcept {
+            std::ostringstream result;
+            result << "{";
+            for (const auto& precision : precisions) {
+                result << precision << "_";
+            }
+            result << "}";
+            return result.str();
+        };
+
+        std::ostringstream result;
+        result << "{";
+        for (const auto& precisions : precisions_pack) {
+            result << to_string(precisions) << "_";
+        }
+        result << "}";
+        return result.str();
+    };
+
+    std::ostringstream result;
+    result << "IN0_" << shapes.first << "_" << test_values.input_types[0] << "_"
+           << "IN1_" << shapes.second << "_" << test_values.input_types[1] << "_"
+           << "IN2_" << test_values.input_types[2]
+           << to_string(test_values.actual.op1_supported_precisions) << "_"
+           << to_string(test_values.actual.op2_supported_precisions) << "_"
+           << test_values.expected.convertion_before_op1.first << "_" << test_values.expected.convertion_before_op1.second << "_"
+           << test_values.expected.convertion_before_op2_1 << "_"
+           << test_values.expected.convertion_before_op2_2.first << "_" << test_values.expected.convertion_before_op2_2.second << "_"
+           << test_values.expected.convertion_after_op2 << "_";
+    return result.str();
+}
+
+TEST_P(PrecisionPropagationTest, CompareFunctions) {
+    disable_rt_info_check();
+
+    const auto param = GetParam();
+    const auto shapes = std::get<0>(param);
+    const auto test_values = std::get<1>(param);
+
+    function = PrecisionPropagationFunction::get<DummyAdd>(test_values.input_types[0],
+                                                           shapes.first,
+                                                           test_values.input_types[1],
+                                                           shapes.second,
+                                                           test_values.input_types[2],
+                                                           test_values.actual.convertion_before_op1,
+                                                           test_values.actual.convertion_before_op2_1,
+                                                           test_values.actual.convertion_before_op2_2);
+
+    const auto target_machine =
+        std::make_shared<DummyPrecisionPropogationTargetMachine>(test_values.actual.op1_supported_precisions,
+                                                                 test_values.actual.op2_supported_precisions);
+    ngraph::snippets::pass::PropagatePrecision(element::f32, target_machine).run_on_model(function);
```

Comment:
Minor: parent class introduces a `manager` field that could be used here:
```suggestion
    manager.register_pass<ngraph::snippets::pass::PropagatePrecision>(element::f32, target_machine);
```

Reply:
I have to run tested transformation here to get ability to serialize result model for debug proposes. Let keep as it is.

Reply:
For debug purposes we can register serialization passes in the same manager. Smth like 
```
manager.register_pass<ov::pass::Serialize>("before.xml", "");
manager.register_pass<ngraph::snippets::pass::PropagatePrecision>(element::f32, target_machine);
manager.register_pass<ov::pass::Serialize>("after.xml", "");
```

Reply:
got it, thanks : ) done

Reply:
actually, test fails with this version. I working on that right now.

Reply:
Maybe we need to add to the manager pass::Validation after ConvertPrecision? Just an idea

Reply:
done, thanks